flags.go

```diff
@@ -142,23 +185,113 @@ func (g *goforit) Enabled(ctx context.Context, name string) (enabled bool) {
 
 	g.flagsMtx.RLock()
 	defer g.flagsMtx.RUnlock()
+	enabled = false
 	if g.flags == nil {
-		enabled = false
+		log.Printf("[goforit] empty flags map")
 		return
 	}
 	lastRefreshTime = g.lastFlagRefreshTime
 	flag := g.flags[name]
 
-	// equality should be strict
-	// because Float64() can return 0
-	if f := g.rand(); f < flag.Rate {
-		enabled = true
+	if !flag.Active {
 		return
 	}
+
+	var mergedProperties = map[string]string{}
+
+	for k, v := range g.defaultTags {
+		mergedProperties[k] = v
+	}
+
+	for k, v := range properties {
+		mergedProperties[k] = v
+	}
+
+	for _, r := range flag.Rules {
+		res, err := r.Rule.Handle(flag.Name, mergedProperties)
+		if err != nil {
+			log.Printf("[goforit] error evaluating rule:\n", err)
```

Comment:
Using `Printf` without any percent-interpolations! That's making some tests fail

Reply:
ah, thats why the builds were failing. great, fixed them up!